### PR TITLE
CTX-4987: Created the Llama2 LoRA template

### PR DESCRIPTION
**Pull request by VukManojlovic**

### CODE REVIEW

File: tasks/llama2-lora/task.yaml [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+    rerquired: true
+  - name: loraRank
+    description: 'Determines the dimensionality of the trainable rank decomposition matrices that
+      are injected into each layer of a Transformer architecture'
+    value: 8
+    data_type: int
+    rerquired: true
+  - name: context
+    description: 'The maximum number of tokens that will be fed into the model at once.
+      This may reduce memory consumption, but can also hurt performence'
```
Comment (dule1322) [inline tasks/llama2-lora/task.yaml:104]: typo in `performance`

File: tasks/llama2-lora/task.yaml [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+    rerquired: false
+  - name: float16
+    description: 'Whether to use float16 if cuda device is selected.
+      Otherwise float32 will be used, i.e. model will not be quantized'
+    value: true
+    data_type: bool
+    required: true
+  - name: testPrompt
+    description: A prompt that will be passed to the model after fine-tuning
+    value: "What is the capital of Paris?"
```
Comment (dule1322) [inline tasks/llama2-lora/task.yaml:116]: Paris is a city so it can't have a capital right?

File: tasks/llama2-lora/task.yaml [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+      alpha adjusts how strongly the low-rank adapted parameters influence the final model behavior,
+      allowing for fine-tuning of the adaptation's impact on model performance.'
+    value: 16
+    data_type: float
+    rerquired: true
+  - name: loraDropout
+    description: 'Dropout randomly disables a fraction of neurons during each training iteration,
+      which helps prevent overfitting by reducing reliance on specific neurons and encourages the
+      network to learn more robust features. Adjusting this parameter affects the level of
+      regularization applied, influencing the model's generalization ability.'
```
Comment (dule1322) [inline tasks/llama2-lora/task.yaml:92]: Add the allowed range of this parameter

**Review by dule1322 — CHANGES_REQUESTED** [review]

File: tasks/llama2-lora/task.yaml (hunk 2 above)
Comment (VukManojlovic) [reply tasks/llama2-lora/task.yaml:116]: That's the point :) This is a test for the model, to see how it responds.

File: tasks/llama2-lora/main.py [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+
+        ######                                        ######
+        # Load data and perform preprocessing as necessary #
+        ######                                        ######
+
+        # Example for loading parquet files from a single sample
+        dataFiles = list(sample.path.glob("*.parquet"))
+        trainingData = load_dataset("parquet", data_files = [str(filePath)for filePath in dataFiles])["train"]
+
+        return trainingData
```
Comment (dule1322) [inline tasks/llama2-lora/main.py:29]: This is loading only the first sample in for loop, why? Also return type of the function is `Any`, can it be changed to a real type?

**Review by dule1322 — CHANGES_REQUESTED** [review]

File: tasks/llama2-lora/main.py (hunk 4 above)
Comment (VukManojlovic) [reply tasks/llama2-lora/main.py:29]: It's part of the "Example for loading parquet files from a single sample".
Return will be changed to Dataset form datasets.

**Review by dule1322 — APPROVED** [review]

**Pull request merged**